adapters/handlers/rest/clusterapi/grpc/server.go

```diff
@@ -34,22 +34,24 @@ type Server struct {
 }
 
 const (
-	PB_OVERHEAD         = 16 * 1024       // 16kB for any extra overhead
-	DEFAULT_MSG_SIZE    = 4 * 1024 * 1024 // 4MB default from grpc
-	INITIAL_CONN_WINDOW = 16 << 20        // 16MB
+	PB_OVERHEAD      = 16 * 1024       // 16kB for any extra overhead
+	DEFAULT_MSG_SIZE = 4 * 1024 * 1024 // 4MB default from grpc
 )
 
 // NewServer creates *grpc.Server with optional grpc.Serveroption passed.
 func NewServer(state *state.State, options ...grpc.ServerOption) *Server {
 	fileCopyChunkSize := state.ServerConfig.Config.ReplicationEngineFileCopyChunkSize
 
 	maxSize := GetMaxMessageSize(fileCopyChunkSize)
+	initialConnWindowSize := maxSize * state.ServerConfig.Config.ReplicationEngineFileCopyWorkers
 
 	o := []grpc.ServerOption{
 		grpc.MaxRecvMsgSize(maxSize),
 		grpc.MaxSendMsgSize(maxSize),
 		grpc.InitialWindowSize(int32(maxSize)),
-		grpc.InitialConnWindowSize(INITIAL_CONN_WINDOW),
+		grpc.InitialConnWindowSize(int32(initialConnWindowSize)),
+		grpc.ReadBufferSize(4 << 20),  // 4 MB
+		grpc.WriteBufferSize(4 << 20), // 4 MB
```

Comment:
wanna put the read and write buffer sizes into consts? bc I can see that those values are being used in 2 places